### DISCUSSION TITLE
error [Service Worker] navigator is undefined

### DISCUSSION THREAD

**Discussion by huanghantao** [coder/code-server#3830] [Q&A]
It's my hosts:

```bash
127.0.0.1	localhost
127.0.0.1 coding.com
127.0.0.1 codinghuang-test1.coding.com
```

When I use localhost to access code-server, there is no problem, but when I use coding.com to access code-server, I get the error.

![image](https://user-images.githubusercontent.com/22836925/126429297-b1e45726-c7f2-4258-bd2d-6e071b3d0255.png)

![image](https://user-images.githubusercontent.com/22836925/126429329-0c5fabce-b6f2-4c5a-8e81-7f56245baaf7.png)

**Comment by jsjoeio** [comment]
It looks like when you use coding.com, you're accessing code-server over an insecure domain, which I believe is the reason the service worker isn't being registered (though it would be nice if we had better error messages). 

Have you followed [the docs](https://coder.com/docs/code-server/v3.11.0/guide#https) for setting up HTTPS?

**Reply by huanghantao** [reply]
Thank you. Is it possible to turn off detection of an insecure domain?

**Reply by huanghantao** [reply]
![image](https://user-images.githubusercontent.com/22836925/126599662-8d71b265-ca12-4d8a-861c-7a680e5c328b.png)
![image](https://user-images.githubusercontent.com/22836925/126599774-6b749ffb-4d12-4d3a-8584-cc5f0a6aaaf5.png)

**Reply by huanghantao** [reply]
Even if I set up the certificate, I still have the same problem.

**Reply by jsjoeio** [reply]
This might be helpful information. May I ask why you're using code-server that way as opposed to using it via `homebrew` or `yarn`/`npm`?

Hmm...maybe `--cert` only works if you're serving code-server on `localhost`? I'm not totally sure. Calling in @code-asher for help

**Reply by huanghantao** [reply]
Because I don't know that I can use yarn, I just found `bin/code-server`.

**Reply by jsjoeio** [reply]
Ah okay. Well if you're open to it, you could try installing with `yarn` (it may or may not improve things but could be worth trying)

**Reply by huanghantao** [reply]
I installed code-server using `rpm -i code-server-3.10.2-amd64.rpm`. Can I start it using Yarn?

**Reply by code-asher** [reply]
The `SSL certificate error occurred` usually means that the browser has determined that your certificate is invalid. There are ways to disable the security although it would be better to use a valid certificate.

**Reply by jsjoeio** [reply]
> Can I start it using Yarn?

You'd have to install it with `yarn`. But let's pause on that idea and consider @code-asher's solution since that will address the root of the issue. 

Is it possible for you to create a valid SSL certificate and try that first?

**Reply by huanghantao** [reply]
> The `SSL certificate error occurred` usually means that the browser has determined that your certificate is invalid. There are ways to disable the security although it would be better to use a valid certificate.

Why are certificates generated through code-server invalid?

**Reply by code-asher** [reply]
Ah are you using `--cert` (without any path)? I wonder if code-server's self-signed certificate is no longer considered valid by browsers (or by Chrome at least). We may need to poke into this and see.

**Reply by huanghantao** [reply]
> Ah are you using `--cert` (without any path)? I wonder if code-server's self-signed certificate is no longer considered valid by browsers (or by Chrome at least). We may need to poke into this and see.

Yes, I was using --cert without any path. My Chrome's info:

```
91.0.4472.106 (x86_64)
```
